cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -1279,7 +1279,7 @@ runQueryLeadershipSchedule (AnyConsensusModeParams cModeParams) network
 
       case mJsonOutputFile of
         Nothing -> liftIO $ printLeadershipScheduleAsText schedule eInfo (SystemStart $ sgSystemStart shelleyGenesis)
-        Just (OutputFile jsonOutputFile) ->
+        Just (File jsonOutputFile) ->
```

Comment:
There are parts of the code where we still unwrap the `FilePath` and use that directly.  All these cases can be converted to use the type-safe functions like `writeLazyByteString` instead.  It isn't done here to keep this PR as small as possible.